src/pytorch_lightning/CHANGELOG.md

```diff
@@ -12,6 +12,11 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 ### Changed
 
 
+### Fixed
+
+- Made checkpointing on train epoch end condition dynamic ([#15300](https://github.com/Lightning-AI/lightning/pull/15300))
```

Comment:
I would consider this a "feature" if we make it dynamic. If you want to fix the linked issue in a bug-fix release, then there should be 2 PRs, one to fix the static issue, and another to add dynamic support